packages/block-library/src/gallery/block.json

```diff
@@ -72,14 +79,33 @@
 			"type": "boolean",
 			"default": true
 		},
+		"linkTarget": {
+			"type": "string"
+		},
 		"linkTo": {
 			"type": "string"
 		},
 		"sizeSlug": {
 			"type": "string",
 			"default": "large"
+		},
+		"allowResize": {
+			"type": "boolean",
+			"default": false
+		},
+		"isGrouped": {
+			"type": "boolean",
+			"default": true
+		},
+		"imageCount": {
+			"type": "number",
+			"default": 0
```

Comment:
We needed to add the imageCount attribute as the save method calculates the default columns from the number of images in the gallery, and at the point of block validation innerBlocks is not available - c9fe911

Reply:
So that's used for adding the `columns-n` classname? I guess for controlling the image width when there aren't enough images for the column count.

Feels like there could be a pure CSS solution to this? Something like this:
https://lea.verou.me/2011/01/styling-children-based-on-their-number-with-css3/

Reply:
Hey @talldan, @glendaviesnz 👋 
Please note that `imageCount` is used on the [react native side](https://github.com/WordPress/gutenberg/pull/31306/files#diff-aa8bcead875dfe40fbceeeebb1050f477de8bd0d70f3213b1d1fe1c6d26b37ffR55) too. Any change to this will also affect the mobile implementation.

Reply:
@glendaviesnz not sure I understand why it cannot rely on counting inner blocks or why block validation is involved?

Reply:
@mtias I am not exactly sure why now either, it was a long time ago  that this was implemented   - I am sure it made perfect sense at the time 😄 In light of the concerns I will take the time to get fully familiar with this part of the code again, and see if there are any pure CSS options for resolving this as @talldan has suggested. @antonis, I will keep you in the loop, and if a different approach is possible I won't make any changes until I talk through the implications with you.

Reply:
Thank you for the ping @glendaviesnz 🙇 
I think we can use the length value of the inner blocks if the `imageCount` is removed.

Reply:
@glendaviesnz hah, I know the feeling. If it's also just the class name that's at odds, we can always render that one server side and not persist it. (It's actually a bit odd to persist it.)

Reply:
@mtias FYI, it turned out we could move the handling of default column numbering to css, so the [imageCount attribute has been removed](https://github.com/WordPress/gutenberg/pull/33677).